config/base.go

```diff
@@ -0,0 +1,97 @@
+package config
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/notaryproject/notation-go/dir"
+)
+
+var (
+	// ConfigPath is the path for config.json
+	ConfigPath string
+	// SigningKeysPath is the path for signingkeys.json
+	SigningKeysPath string
+
+	// fileInfo is the information of config.json
+	fileInfo     *File
+	fileInfoOnce sync.Once
+
+	// signingKeysInfo is the information of signingkeys.json
+	signingKeysInfo     *SigningKeys
+	signingKeysInfoOnce sync.Once
+)
+
+func init() {
+	ConfigPath = dir.Path.Config()
+	SigningKeysPath = dir.Path.SigningKeyConfig()
+}
+
+// Configuration is an interface to manage notation config
+type Configuration interface {
+	Save() error
+}
+
+// Config is the main config struct of notation-go
+type Config struct {
+	*File
```

Comment:
Because we have two kinds of files, namely config.json and signingkeys.json, switching these two looks more reasonable, i.e. 
```
type File struct {
	*Config
	*SigningKeys
}
```

Reply:
Done.